### PR TITLE
Add power function and test to backend

### PR DESCRIPTION
**Pull request by wkostuch**
First step as proposed in #840 to add support for the np.power function.

### CODE REVIEW

**Comment by alewis** [comment]
Before code can be merged, it needs to "build", which means for example that pylint and pytest have to run successfully. You can tell whether your code has built properly by looking at the line marked "continuous-integration/travis-ci/pr", which will have a red X / green checkmark upon failure/success. The same symbols appear next to the PR in the PR list. 

If the code doesn't build, you can click Details on the same line to see which configurations have failed, and then click on any one of them to see the console output from each. In your case the error is quite simple: line 1001 in abstract_backend.py exceeds the maximum character limit of 80 characters.

File: tensornetwork/backends/abstract_backend.py [hunk 1]
```diff
@@ -988,3 +988,16 @@ def deserialize_tensor(self, s: str) -> Tensor:
...
+  def power(self, a: Tensor, b: Tensor) -> Tensor:
+    """
+    Returns the element-wise exponentiation of tensor a raised to tensor b.
+    
+    Args:
+      a: The tensor containing the bases.
+      b: The tensor containing the powers.
+
+    Returns:
+      The tensor that is each element of a raised to the corresponding power in b.
```
Comment (alewis) [inline tensornetwork/backends/abstract_backend.py:1001]: Line too long

File: tensornetwork/backends/abstract_backend.py [hunk 2]
```diff
@@ -988,3 +988,16 @@ def deserialize_tensor(self, s: str) -> Tensor:
     raise NotImplementedError(
         "Backend '{}' has not implemented deserialize_tensor.".format(
             self.name))
+
+  def power(self, a: Tensor, b: Tensor) -> Tensor:
```
Comment (alewis) [inline tensornetwork/backends/abstract_backend.py:992]: b should be allowed to be a scalar as well, in which case the output should be each element in a raised to b.

File: tensornetwork/backends/abstract_backend.py [hunk 3]
```diff
@@ -988,3 +988,16 @@ def deserialize_tensor(self, s: str) -> Tensor:
...
+    Returns the element-wise exponentiation of tensor a raised to tensor b.
+    
+    Args:
+      a: The tensor containing the bases.
+      b: The tensor containing the powers.
+
+    Returns:
+      The tensor that is each element of a raised to the corresponding power in b.
+    """
+    raise NotImplementedError(f"Backend {self.name} has not implemented power.")
```
Comment (alewis) [inline tensornetwork/backends/abstract_backend.py:1003]: Add a newline at the end here

File: tensornetwork/backends/backend_test.py [hunk 4]
```diff
@@ -436,3 +436,8 @@ def test_pivot_not_implemented():
   backend = AbstractBackend()
   with pytest.raises(NotImplementedError):
     backend.pivot(np.ones((2, 2)))
+
+def test_power_not_implemented():
+  backend = AbstractBackend()
+  with pytest.raises(NotImplementedError):
+    backend.power(np.array([1, 2]), np.array([1, 2]))
```
Comment (alewis) [inline tensornetwork/backends/backend_test.py:443]: Add a newline to the end of the file

**Review by alewis — CHANGES_REQUESTED** [review]
Looks good mostly! Fix the small formatting issues here and I'll merge.

File: tensornetwork/backends/abstract_backend.py [hunk 5]
```diff
@@ -988,3 +988,16 @@ def deserialize_tensor(self, s: str) -> Tensor:
     raise NotImplementedError(
         "Backend '{}' has not implemented deserialize_tensor.".format(
             self.name))
+
+  def power(self, a: Tensor, b: Tensor) -> Tensor:
+    """
+    Returns the element-wise exponentiation of tensor a raised to tensor b.
+    
+    Args:
+      a: The tensor containing the bases.
```
[the remark this answers is not in the document]
Comment (wkostuch) [reply tensornetwork/backends/abstract_backend.py:997]: I've made a few changes concerning the shape in the docstring, but I do think I'll be able to express the size conditions more clearly and concisely after the function is written.

[the remark this answers is not in the document]
Comment (alewis) [reply tensornetwork/backends/abstract_backend.py:1003]: SGTM

**Review by alewis — APPROVED** [review]
Ok, looks good. Fix the build and I'll merge.

**Pull request merged**